modules/phase_field/include/auxkernels/ADTotalFreeEnergy.h

```diff
@@ -0,0 +1,37 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "TotalFreeEnergyBase.h"
+
+// Forward Declarations
+
+/**
+ * AD version that uses AD material properties
+ * Total free energy (both the bulk and gradient parts), where the bulk free energy has been defined
+ * in a material and called f_name
+ */
+
+class ADTotalFreeEnergy : public TotalFreeEnergyBase
```

Comment:
This is fine for now. But I'll probably go back to this and combine it into `TotalFreeEnergy` and template it on `is_ad`...

Reply:
So, do i need to fix this in this submission before it is accepted; or are you saying you will work on it later on?